ENGDOCS-999 (#15629)
docker/docs: desktop/dev-environments/create-dev-env.md

---
description: Dev Environments
keywords: Dev Environments, share, collaborate, local

> title: Create a simple dev environment

Reviewer: ```suggestion
title: Create a dev environment
```

---

You can create a dev environment from a:
- Git repository
- Branch or tag of a Git repository
- Subfolder of a Git repository
- Local folder

This did not conflict with any of the local files or local tooling set up on your host. 

## Create a dev environment from a Git repository

> The simplest way to get started with Dev Environments is to create a new environment by cloning the Git repository of the project you are working on. 

Reviewer: ```suggestion
The quickest way to get started with dev environments is to create a new environment by cloning the Git repository of the project you are working on. 
```

    </div>
</div>

To create a dev environment:

1. From **Under Dev Environments** in Docker Dashboard, click **Create**. The **Create a Dev Environment** dialog displays.
2. Select **Get Started** and then copy `https://github.com/dockersamples/single-dev-env.git` and add it to the **Enter the Git Repository** field with **Existing Git repo** as the source.
3. Select **Continue**.

>     This detects the main language of your repository, clones the Git code inside a volume, determines the best image for your Dev Environment, and opens VS Code inside the Dev Environment container.

Reviewer: ```suggestion
    This detects the main language of your repository, clones the Git code inside a volume, determines the best image for your dev environment, and opens VS Code inside the dev environment container.
```

Docker then clones the repository with your specified branch or tag.

## Create a dev environment from a subdirectory of a Git repository

>Note
>

> >Currently, Dev Environments is not able to detect the main language of the subdirectory. You need to define your own base image or compose file in a .docker folder located in your subdirectory. For more information on how to configure, see the [React application with a Spring backend and a MySQL database sample](https://github.com/docker/awesome-compose/tree/master/react-java-mysql) or the [Go server with an Nginx proxy and a Postgres database sample](https://github.com/docker/awesome-compose/tree/master/nginx-golang-postgres). 

Reviewer 2: @nicksieger said he'll review all references to .docker and suggest improvements consistent with what's shipping in 4.13.

> >Currently, Dev Environments is not able to detect the main language of the subdirectory. You need to define your own base image or compose file in a .docker folder located in your subdirectory. For more information on how to configure, see the [React application with a Spring backend and a MySQL database sample](https://github.com/docker/awesome-compose/tree/master/react-java-mysql) or the [Go server with an Nginx proxy and a Postgres database sample](https://github.com/docker/awesome-compose/tree/master/nginx-golang-postgres). 

Reviewer 3: Should we call out `compose-dev.yaml` specifically here, or do we want to explain that elsewhere?

```suggestion
>Currently, Dev Environments is not able to detect the main language of the subdirectory. You need to define your own base image or compose services in a `compose-dev.yaml` located in your subdirectory. For more information on how to configure, see the [React application with a Spring backend and a MySQL database sample](https://github.com/docker/awesome-compose/tree/master/react-java-mysql) or the [Go server with an Nginx proxy and a Postgres database sample](https://github.com/docker/awesome-compose/tree/master/nginx-golang-postgres). 
```

1. From **Dev Environments** in Docker Dashboard, click **Create**. The **Create a Dev Environment** dialog displays.
2. Select **Get Started** and then copy your Git subfolder link into the **Enter the Git Repository** field with **Existing Git repo** as the source.
3. Select **Continue**.

>     This clones the Git code inside a volume, determines the best image for your Dev Environment, and opens VS Code inside the Dev Environment container.

Reviewer: ```suggestion
    This clones the Git code inside a volume, determines the best image for your dev environment, and opens VS Code inside the environment container.
```

5. To launch the application, run the command `make run` in your terminal. This opens an http server on port 8080. Open [http://localhost:8080](http://localhost:8080) in your browser to see the running application.

## Create a dev environment from a local folder

1. From **Dev Environments** in Docker Dashboard, click **Create**. The **Create a Dev Environment** dialog displays.
2. Select **Get Started** and then choose **Local Folder** as the source.
3. Select **Select directory** to open the root of the code that you would like to work on.
4. Select **Continue**.

>     This detects the main language of your local folder, creates a Dev Environment using your local folder, and bind-mounts your local code in the Dev Environment. It then opens VS Code inside the Dev Environment container.

Reviewer: ```suggestion
    This detects the main language of your local folder, creates a dev environment using your local folder, and bind-mounts your local code in the dev environment. It then opens VS Code inside the environment container.
```
